builtin/providers/vsphere/resource_vsphere_virtual_machine_test.go

```diff
@@ -33,7 +33,7 @@ func TestAccVSphereVirtualMachine_basic(t *testing.T) {
 	if v := os.Getenv("VSPHERE_DATASTORE"); v != "" {
 		datastoreOpt = fmt.Sprintf("        datastore = \"%s\"\n", v)
 	}
-	template := os.Getenv("VSPHERE_TEMPLATE")
+	cloneSource := os.Getenv("VSPHERE_CLONE_SOURCE")
```

Comment:
I actually did think about few months ago to refactor the vsphere module. I just wasn't sure if you guys would be happy with that much of an interference. 😛 

The refactoring would mostly be in a proper struct based code, proper godocs and interfacing where needed. Just the "proper" structure. Not saying anything is wrong with the code, it's just to get rid of all the passing-around variables to all the different functions, with huge signature, etc.